acra/configuring-maintaining/tls/crl.md

```diff
@@ -0,0 +1,99 @@
+---
+weight: 2
+title: CRL
+bookCollapseSection: true
+---
+
+# TLS certificate validation using CRL
+
+Only [CRL v1](https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.1) with basic fields is currently supported.
+This means Acra will be able to download CRL, verify it, and search for a certificate serial number there.
+But more advanced features like [CRL extensions](https://datatracker.ietf.org/doc/html/rfc5280#section-5.2)
+(including [delta CRLs](https://datatracker.ietf.org/doc/html/rfc5280#section-5.2.4)) are not handled yet.
+
+CRL-related flags and their description. Work in `acra-connector`, `acra-server` and `acra-translator`.
+
+* `--tls_crl_url=<url>`
+
+  URL of the Certificate Revocation List (CRL) to use.
+  Empty by default.
+
+  Can be either `http://` or `file://` (for local files).
+  When using local file, Acra will simply read the file and won't monitor filesystem for changes afterwards.
+  Usual caching rules apply (see `--tls_crl_cache_time`).
+
+  For `acra-server` it will work like `--tls_crl_client_url` and `--tls_crl_database_url`
+  passed simultaneously with same value.
+
+  For `acra-connector` and `acra-translator` (that can only work as TLS clients)
+  it will set CRL URL for validation of certificates sent by server.
+
+* `--tls_crl_client_url=<url>`
+
+  CRL URL for incoming TLS connections to check client certificates.
+  Empty by default.
+  {{< hint warning >}}
+  Only for `acra-server`
+  {{< /hint >}}
+
+* `--tls_crl_database_url=<url>`
+
+  CRL URL for outcoming TLS connections to check database certificates.
+  Empty by default.
+  {{< hint warning >}}
+  Only for `acra-server`
+  {{< /hint >}}
+
+* `--tls_crl_from_cert=<policy>`
+
+  How to treat CRL URL described in certificate itself
+
+  * `use` — try URL(s) from certificate after the one from configuration (if set)
+  * `trust` — try first URL from certificate, if it does not contain checked certificate, stop further checks
+  * `prefer` — (default) try URL(s) from certificate before the one from configuration (if set)
+  * `ignore` — completely ignore CRL URL(s) specified in certificate
+
+  "URL from configuration" above means the one configured with `--tls_crl_*_url` flags.
+
+* `--tls_crl_check_only_leaf_certificate={true|false}`
+
+  This flag controls behavior of validator in cases when certificate chain contains at least one intermediate certificate.
```

Comment:
maybe we should note that last ICA before leaf certificate should be trusted and this option may be used in cases of long certificate chain with several intermediate CAs that may not be placed close to validation side? and same for ocsp

Reply:
Done. Is new description good enough?